c++/src/ColumnReader.cc

```diff
@@ -412,10 +412,11 @@ namespace orc {
     nanoRle->seek(positions.at(columnId));
   }
 
+  template<TypeKind COLUMN_KIND, bool IS_LITTLE_ENDIAN>
```

Comment:
Although I know that this pattern is used before, this still looks misleading. For example, this PR is introducing the following which is very confusing because both one looks like CONSTANT.
```
if (COLUMN_KIND == FLOAT) {
```

Reply:
Is this the best practice for us to use as Apache ORC C++ coding standard, @stiga-huang ? It's up to you.

Reply:
Thanks for raising this question! I think I should rename `COLUMN_KIND` to `columnKind` and `IS_LITTLE_ENDIAN` to `isLittleEndian` since they are not type names.